### PR TITLE
Add nginx

### LINKED ISSUE #8: Add nginx
Currently, we serve our application directly by exposing the gunicorn from our container. Due to this, we are not able to serve our static file as Django doesn't support and recommend serving static files in production on its own. We need to use services like apache or nginx.

- [ ] Install Nginx inside the container
- [ ] Create `nginx.conf`
    - [ ] Serve static file under tha url written in `settings.py`
    - [ ] proxy pass rest of the request to the gunicorn running on port 8000
- [ ] update entrypoint script
    - [ ] enable our nginx conf
    - [ ] run collect static before spinning up the server

### PR DESCRIPTION
**Pull request by prateek2211**
fix #8 
* [X] Install Nginx inside the container
* [X] Create nginx.conf
   * [X] Serve static file under tha url written in settings.py
   * [X] proxy pass rest of the request to the gunicorn running on port 8000
* [X] update entrypoint script
   * [X] enable our nginx conf
   * [X] run collect static before spinning up the server

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ WORKDIR /usr/src/poker
 COPY Pipfile Pipfile
 COPY Pipfile.lock Pipfile.lock
 RUN pipenv install --system
+COPY nginx.conf /etc/nginx/sites-enabled/poker
```
Comment (jatinkatyal13) [inline Dockerfile]: copy the file to `/etc/nginx/sites-available` and create a symlink in `/etc/nginx/sites-enabled`

File: poker/settings.py [hunk 2]
```diff
@@ -13,47 +13,47 @@
...
 INSTALLED_APPS = [
-    'oneauth',
-    'cron',
-    'django.contrib.admin',
-    'django.contrib.auth',
-    'django.contrib.contenttypes',
-    'django.contrib.sessions',
-    'django.contrib.messages',
-    'django.contrib.staticfiles',
+  'oneauth',
```
Comment (jatinkatyal13) [inline poker/settings.py]: remove unnecessary changes

File: start.sh [hunk 3]
```diff
@@ -2,4 +2,6 @@
 
 /usr/sbin/crond
 python manage.py migrate
+nginx
+echo 'yes' | python manage.py collectstatic
```
Comment (jatinkatyal13) [inline start.sh]: remove echo

**Review by jatinkatyal13 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (prateek2211) [reply Dockerfile]: done

File: poker/settings.py (hunk 2 above)
Comment (prateek2211) [reply poker/settings.py]: by unnecessary changes, u meant formatting changes. Right?

File: start.sh (hunk 3 above)
Comment (prateek2211) [reply start.sh]: I tried but this doesn't work
`yes | python manage.py collectstatic `

Comment (prateek2211) [reply start.sh]: Though ` yes yes|python manage.py collectstatic` works.
Should I use this instead?

File: poker/settings.py (hunk 2 above)
Comment (jatinkatyal13) [reply poker/settings.py]: yes

File: start.sh (hunk 3 above)
Comment (jatinkatyal13) [reply start.sh]: What is the error thrown by Django?
It should most probably because while the postgres container is booting up, it tries to connect to it and fails. You can try using scripts like [wait-for-it](https://github.com/vishnubob/wait-for-it/blob/master/wait-for-it.sh) to wait for the postgres container to boot up. This script should go as the entrypoint of the container

Comment (prateek2211) [reply start.sh]: python manage.py collectstatic needs yes or no confirmation,if we dont enter yes or no, then it doesn't run.
Using `yes` just outputs `y` , while using echo yes, or `yes yes` outputs yes

Probably, I didn't know get you?

Comment (jatinkatyal13) [reply start.sh]: add  --noinput flag

Comment (prateek2211) [reply start.sh]: ok

**Review by jatinkatyal13 — APPROVED** [review]

**Comment by jatinkatyal13** [comment]
Nice Job 🚀
